ledger/src/blockstore.rs

```diff
@@ -7756,20 +7729,17 @@ pub mod tests {
             ));
         }
 
-        // Establish a baseline that works
-        coding_shred.set_index(coding_shred.index() + 1);
         assert!(Blockstore::should_insert_coding_shred(
-            &coding_shred,
+            &code_shreds[1],
             max_root
         ));
 
         // Trying to insert value into slot <= than last root should fail
         {
-            let mut coding_shred = coding_shred.clone();
-            coding_shred.set_slot(max_root);
+            let coding_shred = coding_shred.clone();
             assert!(!Blockstore::should_insert_coding_shred(
                 &coding_shred,
-                max_root
+                max_root + 1
```

Comment:
why +1 the root instead of flooring the shred slot?

Reply:
we can not modify merkle shreds as it would invalidate merkle tree.

Reply:
Why not just make a new shred at slot `max_root` ?

Reply:
Or, to make it more clear and avoid need to create extra shreds, we could pass `coding_shred.slot()` instead of `max_root + 1`.

Reply:
> Why not just make a new shred at slot `max_root` ?

We'd have to make 64 of them 😢 .

> Or, to make it more clear and avoid need to create extra shreds, we could pass coding_shred.slot() instead of max_root + 1.

I'll do that, thanks!